migration-engine/ARCHITECTURE.md

```diff
@@ -117,6 +118,41 @@ On a high level, migrate deploy _exclusively_:
   looking at the `_prisma_migrations` table in the target database.
 - Run the migrations that have not been applied yet, in chronological order.
 
+`prisma migrate deploy` is the command meant to be used for _unattended_
+migrations (as part of CI). As such, it should be as reliable, predictable and
+deterministic as possible.
+
+### Why does `prisma migrate deploy` not warn when a migration is in the migrations table but not in the migrations folder?
+
+We don't want to warn if an already applied migration is missing from the
+migration directory because it would prevent squashing migrations and
+deployment from not-properly-rebased branches.
+
+In general, by design, `deploy` errs on the side of not standing in the way of
+deploying your migrations.
```

Comment:
Can we use another word for `errs` I first thought it was a typo for errors 😓

Reply:
It's a rather common expression. I can't easily find synonyms (https://www.merriam-webster.com/dictionary/err%20on%20the%20side%20of doesn't list any), but I'm happy to oblige if there's one.

Reply:
```suggestion
In general, by design, `deploy` takes the safe approach of not standing in the way of
deploying your migrations.
```
maybe works too

Reply:
I'm trying hard to come up with a different phrasing for this. The best I have is "deploy is designed to not stand in the way of deploying your migration", but it's not exactly what I mean. There's a tradeoff and there is no right choice and no safe approach, we have to balance our design, and the balance tends to go a bit more on the "deploy anything" than the "all checks enabled in strictest mode" direction.

Reply:
The currently merged doc with `errs` doesn't help to understand this though